src/llmcompressor/transformers/sparsification/sparse_model.py

```
@@ -1,174 +1,29 @@
-    1. Decompress a compressed model
-    2. Initialize any saved recipes
-    3. Wrap the `save_pretrained` method to allow saving as a compressed model
-
-    :param hf_model_class: Model class to wrap
-    :return: Wrapped model class
-    """
-
-    # Add the from_pretrained class method
-    @classmethod
-    def from_pretrained(
-        cls,
-        pretrained_model_name_or_path,
-        run_compressed: bool = False,
-        recipe: Optional[Union[str, Path]] = None,
-        *model_args,
-        **kwargs,
-    ) -> PreTrainedModel:
-        """
-        A wrapper around the PreTrainedModel.from_pretrained method
-
-        :param pretrained_model_name_or_path: the name of or path to the model to load
-        :param recipe: the path to the recipe file to apply to the model. Can be a
-            string or Path object. If None, a recipe will be searched for in the
-            pretrained_model_name_or_path directory and applied if found
-        :return the created model for causal language modeling
-        """
-
-        def skip(*args, **kwargs):
-            pass
-
-        # Skip the initializer step. This accelerates the loading
-        # of the models, especially for the quantized models
-        torch.nn.init.kaiming_uniform_ = skip
-        torch.nn.init.uniform_ = skip
-        torch.nn.init.normal_ = skip
-
-        pretrained_model_name_or_path = (
-            pretrained_model_name_or_path.as_posix()
-            if isinstance(pretrained_model_name_or_path, Path)
-            else pretrained_model_name_or_path
-        )
-
-        pretrained_model_name_or_path = download_model_directory(
-            pretrained_model_name_or_path, **kwargs
-        )
-
-        # instantiate compressor from model config
-        compressor = ModelCompressor.from_pretrained(
-            pretrained_model_name_or_path, **kwargs
+class SparseAutoModelForCausalLM:
+    def from_pretrained(*args, **kwargs):
+        logger.warning(
```

Since this class is lot actually usable, this should be an error rather than a warning

---

Warnings should be used in cases where it's okay to continue. Unless you subclass AutoModel, this is not a case where the user can continue.

---

I agree, let's raise an error instead

---

you can use this code, we can completely delete the pathway later